Sources/Down/AST/Styling/Stylers/DownStyler.swift

```diff
@@ -218,6 +219,28 @@ open class DownStyler: Styler {
             .link: url,
             .foregroundColor: colors.link])
     }
+    
+    private func styleGenericImg(in str: NSMutableAttributedString, url: URL) {
+        let semaphore = DispatchSemaphore(value: 0)
+        URLSession.shared.dataTask(with: url) { data, _, _ in
+            if let data = data {
+                let image1Attachment = NSTextAttachment()
+                
+                #if os(macOS)
+                image1Attachment.image = NSImage(data: data)
+                #else
+                image1Attachment.image = UIImage(data: data)
+                #endif
+                let image1String = NSAttributedString(attachment: image1Attachment)
+                
+                str.setAttributedString(image1String)
+            }
+            
+            semaphore.signal()
+        }.resume()
+        
+        semaphore.wait()
```

Comment:
This effectively makes the loading of the image a synchronous task, which means we will block rendering the attributed string for each image it contains. While I appreciate the simplicity of it, I'm concerned that it will slow things down, especially with poor network conditions.

It would be great to load images asynchronously. I haven't yet thought about how that would look like, but would be happy to brainstorm some ideas together.

One quick idea is to start fetching the image (without waiting for it's completion), then mark the string with a control attribute that can be used to later identify it, and in the completion of the image fetch use that marker to find the substring than needs to be replaced with the image attachment.

Reply:
@johnxnguyen I am faced with the fact that if we make this task asynchronous, then down.to The AttributedString terminates before all images are loaded, so the images are not displayed on the screen. There is an idea to try to make getting a ready-made AttributedString via callback

Reply:
@johnxnguyen Or the second option that came to my mind: this is a delegate. After loading the image, call using the text update delegate, BUT this solution will force Down users to write additional code :cry:

Reply:
Found a better solution =)